code/modules/mob/living/living_defense.dm

```diff
@@ -91,45 +91,85 @@
 /mob/living/proc/is_ears_covered()
 	return null
 
-/mob/living/bullet_act(obj/projectile/hitting_projectile, def_zone, piercing_hit = FALSE)
+/mob/living/bullet_act(obj/projectile/proj, def_zone, piercing_hit = FALSE)
 	. = ..()
-	if(. != BULLET_ACT_HIT)
+	var/blocked = check_projectile_armor(def_zone, proj, is_silent = TRUE)
+	if(blocked >= 100)
+		if(. == BULLET_ACT_HIT && proj.is_hostile_projectile())
+			apply_projectile_effects(proj, def_zone, blocked)
 		return .
-	if(!hitting_projectile.is_hostile_projectile())
-		return BULLET_ACT_HIT
 
-	// we need a second, silent armor check to actually know how much to reduce damage taken, as opposed to
-	// on [/atom/proc/bullet_act] where it's just to pass it to the projectile's on_hit().
-	var/armor_check = check_projectile_armor(def_zone, hitting_projectile, is_silent = TRUE)
+	var/hit_limb_zone = check_hit_limb_zone_name(def_zone)
+	var/organ_hit_text = ""
+	if (hit_limb_zone)
+		organ_hit_text = "in \the [parse_zone_with_bodypart(hit_limb_zone)]"
 
+	switch (proj.suppressed)
+		if (SUPPRESSED_QUIET)
+			to_chat(src, span_userdanger("You're shot by \a [proj] [organ_hit_text]!"))
+		if (SUPPRESSED_NONE)
+			visible_message(span_danger("[src] is hit by \a [proj] [organ_hit_text]!"), \
+					span_userdanger("You're hit by \a [proj] [organ_hit_text]!"), null, COMBAT_MESSAGE_RANGE)
+			if(is_blind())
+				to_chat(src, span_userdanger("You feel something hit you [organ_hit_text]!"))
```

Comment:
These feedback messages should *also* be behind `. == BULLET_ACT_HIT`, as otherwise you may have a bullet which you block via comsig falsely reporting that it hits you